File: 326.java
class Solution {
    public boolean isPowerOfThree(int n) {
        //递归
        // if(n == 1) return true;
        // if(n == 0) return false;
        // return n%3 == 0 ? isPowerOfThree(n/3) : false;
        
        //循环
        while(n > 0 && n%3 == 0) {
            n /= 3;
        }
        return n == 1;
    }
}

File: 326.go
func isPowerOfThree(n int) bool {
	//递归
	// if n == 1 {
	//     return true
	// }
	// if n <= 0 || n % 3 != 0 {
	//     return false
	// }
	// return isPowerOfThree(n/3)

	//循环
	for ; n > 0 && n%3 == 0; n /= 3 {
	}
	return n == 1
}